### PR TITLE
Blocks: Usage difficulties due to theme styling

### LINKED ISSUE #5239: Blocks: Usage difficulties due to theme styling
## Bug Description

- Bug: On theme Twenty Sixteen 2.2 (stories-lab.dev), Gutenblock has ~0 height with left or right alignment (due to CSS float)
- Impact: Hard to select the block (need to use arrow keys or block navigation)

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* Frontend: Force `display: block;` for `.web-stories-embed` element
* Editor: Don't set `width: 100%` for `.web-stories-embed.alignleft, .web-stories-embed.alignright`

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

The align functionality of blocks, using floats to align right / left. Right and left float can result in issues with 0 heights. 

## Relevant Technical Choices

- In the CMS, the float without a fixed, means that the block itself results in 0 height, meaning it can not be selected. Adding a simple clearfix, seems to workaround this issue. 
- Add margin-left/right to align left/right. 

## Testing Instructions
1. Click Themes 
2. Activate Twenty sixteen theme. 
3. Create a new post. 
4. Create story embed block. 
5. Input url. 
6. Align right or left. 
7. Save post. 
8. View post on FE. 

This should be tested again all core themes. 

---

Fixes #5239

### CODE REVIEW

File: includes/assets/embed.css [hunk 1]
```diff
@@ -33,6 +28,7 @@
 }
 
 .web-stories-embed:not(.web-stories-embed-amp) .wp-block-embed__wrapper {
+  min-width: var(--width);
```
Comment (swissspidy) [inline includes/assets/embed.css]: With this change, does the block still resize the same on a smaller screen?

File: assets/src/story-embed-block/block/edit.css [hunk 2]
```diff
@@ -1,5 +1,11 @@
 /* Loading Indicator */
```
Comment (swissspidy) [inline assets/src/story-embed-block/block/edit.css]: Need to move that comment to the right placce now, FWIW

File: assets/src/story-embed-block/block/edit.css [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
-
+.block-editor-block-list__block[data-type='web-stories/embed'][data-align='right']
+  .wp-block,
+.block-editor-block-list__block[data-type='web-stories/embed'][data-align='left']
+  .wp-block {
+  height: auto;
+  overflow: hidden;
```
Comment (swissspidy) [inline assets/src/story-embed-block/block/edit.css:3]: Does this cause any visual changes?

File: assets/src/story-embed-block/block/edit.css (hunk 2 above)
Comment (spacedmonkey) [reply assets/src/story-embed-block/block/edit.css]: Fixed.

File: assets/src/story-embed-block/block/edit.css (hunk 3 above)
Comment (spacedmonkey) [reply assets/src/story-embed-block/block/edit.css:3]: not that I can see.

**Comment by spacedmonkey** [comment]
I think this solution is not sound. I am going to close this PR and start again.

**Comment by spacedmonkey** [comment]
I tested it the 10 core themes. This simple fix seems work. But as you can see from the example, not in all cases perfectly. Some of issues are issues with the themes themselves, 2020, is a good example as you get the same result with a aligned image. 

localhost_8899_test-22 (11)
localhost_8899_test-22 (10)
localhost_8899_test-22 (9)
localhost_8899_test-22 (8)
localhost_8899_test-22 (7)
localhost_8899_test-22 (6)
localhost_8899_test-22 (5)
localhost_8899_test-22 (4)
localhost_8899_test-22 (3)
localhost_8899_test-22 (2)

**Review by swissspidy — APPROVED** [review]
There might still be some minor editor-vs-frontend differences in some themes, but this fixes the most pressing issues well.

**Pull request merged**